sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsCachingReader.scala

```diff
@@ -106,8 +106,16 @@ class RapidsCachingReader[K, C](
           require(
             blockManagerId.topologyInfo.isDefined &&
               blockManagerId.topologyInfo.get
-                .startsWith(s"${RapidsShuffleTransport.BLOCK_MANAGER_ID_TOPO_PREFIX}="),
-            s"Attempting to handle non-rapids enabled blocks from $blockManagerId")
+                .startsWith(s"${RapidsShuffleTransport.BLOCK_MANAGER_ID_TOPO_PREFIX}="), {
+              val enabledHint = if (!rapidsConf.shuffleTransportEnabled) {
```

Comment:
I'm still confused why this is even a setting at all.  UCX is lazily loaded, so in the single-executor case the code doesn't try to do a remote shuffle and therefore doesn't load UCX.  What is the use-case where this property makes sense?

Reply:
If we don't have this flag, then we will load UCX at block _write_ time. We are going to want to know then what the port for UCX is, so this creates the UCX class and tries to load those libraries.

I could split up having this port created, and actually initiating the library. But this is the reason for the flag. If it is set to false, we skip the "getting the port" step, and so our blocks don't carry this information when we commit map status.